torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1009,9 +1009,28 @@ void ProcessGroupNCCL::broadcastUniqueNCCLID(
         reinterpret_cast<uint8_t*>(ncclID) + NCCL_UNIQUE_ID_BYTES);
     store_->set(storeKey, vec);
   } else {
-    auto vec = store_->get(storeKey);
-    TORCH_CHECK(vec.size() == NCCL_UNIQUE_ID_BYTES);
-    std::memcpy(ncclID, vec.data(), vec.size());
+    std::string exceptionMsg = c10::str(
```

Comment:
Shall we only construct the string if we have encountered an error for efficiency reasons?